crates/red_knot_python_semantic/src/program.rs

```diff
@@ -142,11 +144,31 @@ pub enum PythonPath {
     /// `/opt/homebrew/lib/python3.X/site-packages`.
     ///
     /// [`sys.prefix`]: https://docs.python.org/3/library/sys.html#sys.prefix
-    SysPrefix(SystemPathBuf),
+    SysPrefix(SysPrefixPath),
 
     /// Resolved site packages paths.
     ///
     /// This variant is mainly intended for testing where we want to skip resolving `site-packages`
     /// because it would unnecessarily complicate the test setup.
     KnownSitePackages(Vec<SystemPathBuf>),
 }
+
+impl PythonPath {
+    pub fn find_virtual_env(system: &dyn System) -> Option<Self> {
+        let virtual_env = std::env::var("VIRTUAL_ENV").ok();
+        if let Some(virtual_env) = virtual_env {
+            tracing::debug!("Found virtual environment at {:?}", virtual_env);
+            return SysPrefixPath::new(virtual_env, SysPrefixPathOrigin::VirtualEnvVar, system)
+                .map(Self::SysPrefix)
+                .ok();
+        }
+        None
```

Comment:
You can simplify this a little by short-circuiting immediately using `?` if `std::env::var("VIRTUAL_ENV").ok()` returns `None`:

```suggestion
        let virtual_env = std::env::var("VIRTUAL_ENV").ok()?;
        tracing::debug!("Found virtual environment at {:?}", virtual_env);
        SysPrefixPath::new(virtual_env, SysPrefixPathOrigin::VirtualEnvVar, system)
            .ok()
            .map(Self::SysPrefix)
```

I think it might also be good to say in the tracing message that we found the virtual environment from the `VIRTUAL_ENV` variable rather than `--python` or anything else -- that could be useful information when debugging. So maybe something like this?

```suggestion
        let virtual_env = std::env::var("VIRTUAL_ENV").ok()?;
        tracing::debug!("Found virtual environment at {:?} from `VIRTUAL_ENV` environment variable", virtual_env);
        SysPrefixPath::new(virtual_env, SysPrefixPathOrigin::VirtualEnvVar, system)
            .ok()
            .map(Self::SysPrefix)
```